File: backend/app/schemas/order.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel
from app.schemas.product import Product

class OrderItemBase(BaseModel):
    product_id: int
    variant_id: Optional[int] = None
    quantity: int
    unit_price: float

class OrderItem(OrderItemBase):
    id: int
    order_id: int
    product_name: str
    sku: str
    total_price: float
    product: Optional[Product] = None

    class Config:
        from_attributes = True

class OrderBase(BaseModel):
    address_id: int
    notes: Optional[str] = None
    payment_method: str = "cod"

class OrderCreate(OrderBase):
    pass

class Order(OrderBase):
    id: int
    order_number: str
    user_id: int
    status: str
    payment_status: str
    subtotal: float
    shipping_fee: float
    tax_amount: float
    total_amount: float
    created_at: datetime
    items: List[OrderItem]

    class Config:
        from_attributes = True

File: frontend/src/types/order.ts
import { Address } from "./auth"
import { Product } from "./product"

export type OrderStatus = "pending" | "processing" | "shipped" | "delivered" | "cancelled"
export type PaymentStatus = "pending" | "paid" | "failed"

export interface OrderItem {
    id: number
    order_id: number
    product_id: number
    variant_id?: number
    product?: Product
    product_name: string
    sku: string
    quantity: number
    unit_price: number
    total_price: number
}

export interface Order {
    id: number
    order_number: string
    user_id: number
    address_id: number
    status: OrderStatus
    payment_status: PaymentStatus
    payment_method?: string
    subtotal: number
    shipping_fee: number
    tax_amount: number
    total_amount: number
    notes?: string
    created_at: string
    items: OrderItem[]
    shipping_address?: {
        label?: string
        full_address: string
        city: string
        state: string
        pincode: string
    }
}
